### PR TITLE
Edit codelist-coverage.json for subnationalCoverage

### PR DESCRIPTION
**Pull request by edugomez**
- Updated codelist for ```subnationalCoverage``` as for ISO-3166:2.
e.g. https://www.iso.org/obp/ui/#iso:code:3166:AF, using English names or Roman transliteration most commonly used in English.

-  Updated ```list-schema.json``` accordingly.

- Changed prefixes for subnational coverage for CN and US lists using underscores (so e.g. ```CN_35``` is now ```CN-35```)
- Changed prefixes for AE lists using underscores and changed name of the files from using underscores to using dashes.
- Fix minor bug in ```update.py``` script.

[jsonschema_test.py](https://github.com/org-id/register/blob/master/tests/jsonschema_tests.py) doesn't seem to be validating that correct ```enum```s are used for ```subnationalCoverage``` items in the lists, e.g. in master ```US-ME ```in [us-me-cr.json](https://github.com/org-id/register/blob/master/lists/us/us-me-cr.json#L45 ) is ```US_ME``` in _list-schema.json_

Notes on subnational divisions: 

1) Subnational divisions for cases when they are just minor territorial classifications are not included; for  example, Andorra and Malta divisions are codified by ISO but are not included in the codelist as they are small municipal divisions in very small countries. However, on second thoughts this could be the wrong approach. Could we have a list covering just a number of small municipalities or even parishes (I doubt it, but I'm not sure)?      
2) For countries with parent divisions, only the parent divisions are included. Again, this may not be the right approach to cater for all (future) cases. Thoughts?

### CODE REVIEW

**Comment by BobHarper1** [comment]
1. > Subnational divisions for cases when they are just minor territorial classifications are not included; for example,

It's possible that there is some division of registration at levels lower than the entire municipality (e.g. Malta has a number of councils).

> However, on second thoughts this could be the wrong approach. Could we have a list covering just a number of small municipalities or even parishes (I doubt it, but I'm not sure)?

I think just to include municipality subdivisions as part of the 3166-2 codes with the rest, if that is possible?

2. > For countries with parent divisions, only the parent divisions are included. Again, this may not be the right approach to cater for all (future) cases. Thoughts?

I'm not sure what the decision should be. At the moment, 'GB' has England, Wales, Scotland and NI as the parent division included, meaning there is no division lower than this (e.g. council) in the org-id codelist https://www.iso.org/obp/ui/#iso:code:3166:GB While I don't think there are any registers we're currently interested in that have been below those country-level divisions (and the UK is probably an unusual example), I'm not sure that will always be the case in other examples/subnational areas.

**Comment by timgdavies** [comment]
> 2. For countries with parent divisions, only the parent divisions are included. Again, this may not be the right approach to cater for all (future) cases. Thoughts?

I think this is the right approach. The sub-national divisions are mainly for the front-end interface drill-down. 

We should only explore adding more granular options when we find a user is struggling to find the list they need because they are encountering too many. We might find that adding free-text search instead of more detailed faceting could meet those discovery use cases.

**Comment by timgdavies** [comment]
> Could we have a list covering just a number of small municipalities or even parishes (I doubt it, but I'm not sure)?

We could have such lists - but we wouldn't need to classify their exact subnational extent in our data. This could exist in descriptions.

**Comment by timgdavies** [comment]
@edugomez Thanks for all the work on this - and sorry for the delay in reviewing.

I've just pushed some small changes to the branch for you to check - and then I'd be happy to merge this.

* (1) I've updated the list codes in US and AE (all of which I think are not yet live codes, just imports from Open Corporates) so that the code only ever follows a 'ISO2DIGIT-PREFIX' pattern (in line with the approach we describe at http://docs.org-id.guide/en/latest/metadata/);
* (2) I've updated `update.py` so that it includes the country name within the label in `enum_titles` as this will make our editing interface a lot easier to use. I don't think this should affect the front-end application? 

Through this process I've noted that our Canada codes (CA_PE-PEI_IPE etc.) are not following the proper rules for using only an ISO 2-digit code to start their codes so I will look at getting these updated through a separate PR and process.

**Comment by timgdavies** [comment]
I also couldn't get `jsonschema_test.py` to report on enums. Perhaps something we should dig into more. I'm thinking we might also want to add a regex for `code` so that the Canada codes issue I noted above doesn't recur. Will open issue for that.

File: lists/ae/ae-aj-cr.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
     "coverage": [
         "AE"
     ],
-    "code": "AE-AJ-CR",
+    "code": "AE-AJ_CR",
```
Comment (edugomez) [inline lists/ae/ae-aj-cr.json]: @timgdavies [the handbook](http://docs.org-id.guide/en/latest/metadata/#list-name-code) states that underscores are used for multilingual codes. However, here the part before the underscore is a subnational division code. Is that correct?

File: schema/update.py [hunk 2]
```diff
@@ -11,7 +11,10 @@ def update_schema(schema,filename,codelist,target,array=True):
         filedata = json.loads(file.read(),object_pairs_hook=collections.OrderedDict)
         for code in filedata[codelist]:
             enum.append(code['code'])
-            enum_titles.append(code['title']['en'])
+            try:
+                enum_titles.append(code['countryCode'] + " > " + code['title']['en'])
```
Comment (edugomez) [inline schema/update.py]: :+1:

**Pull request merged**